cmd/adapter/main.go

```diff
@@ -105,6 +108,12 @@ func (a *Adapter) makeProvider(ctx context.Context) (provider.ExternalMetricsPro
 	cfg.Burst = adapterClientRequestBurst
 	cfg.DisableCompression = disableCompression
 
+	// register grpc metrics
+	var clientMetrics *grpcprom.ClientMetrics
+	if enablePrometheusMetrics {
+		clientMetrics = metricscollector.NewPromClientMetrics()
+	}
```

Comment:
we should include the new metrics within `getMetricHandler`
https://github.com/kedacore/keda/blob/7ee87810685307cb7d236a65c509b8494eb8076d/cmd/adapter/main.go#L144-L159

Could you add them directly to the handler instead of in the file `prommetrics.go`? that file is for operator metrics

Reply:
I did this a bit different, I put the interceptor in its own helper as it's  used in a different spot than the metrics handler